### PR TITLE
docs: update changelog for 0.8.0rc1

### PR DESCRIPTION
**Pull request by henryiii**
Preparing for a release (mentioned in https://github.com/FFY00/python-pyproject-metadata/pull/83).

### CODE REVIEW

**Comment by henryiii** [comment]
(Personally, I'd be fine with a RC release first to make sure PDM & scikit-build-core (and others?) are fine.)

File: pyproject_metadata/__init__.py [hunk 1]
```diff
@@ -24,7 +24,7 @@
 import packaging.version
 
 
-__version__ = '0.7.1'
+__version__ = '0.8.0'
```
Comment (henryiii) [inline pyproject_metadata/__init__.py:27]:
```suggestion
__version__ = '0.8.0rc1'
```

**Comment by henryiii** [comment]
@FFY00 @pradyunsg gentle reminder. :) (From https://github.com/FFY00/python-pyproject-metadata/pull/83#issuecomment-1911474463)

File: CHANGELOG.rst [hunk 2]
```diff
@@ -2,6 +2,19 @@
 Changelog
 +++++++++
 
+0.8.0
+=====
+
+- Preserve the user-specified name style in the metadata. ``.canonical_name`` added to get the normalized name
+- Require "version" in the dynamic table if unset (following PEP 621)
+- Support extras using markers containing "or"
```
Comment (henryiii) [inline CHANGELOG.rst:10]:
```suggestion
- Fix email-author listing for names containing commas
```

**Comment by rooterkyberian** [comment]
@henryiii AFAIK pdm won't be affected since they are explicitly vendoring this library, so they have to manually port changes every time. 

On that topic - I just prepared such porting PR and it seem now maintainer of pdm is blocking vendoring new version waiting for official release ( https://github.com/pdm-project/pdm-backend/pull/220#issuecomment-2028973672 ), so my questions would be:
* when this release can happen?
* can it include fixes from current master (I'm only really intrested in [comma fix](https://github.com/pypa/pyproject-metadata/commit/d263e299cc019a3898e5b80d6bdd60c0367fa6ce)  )
* anything I can do to help?

**Comment by henryiii** [comment]
@FFY00 and @pradyunsg are the only ones that can make a release or set up trusted publishing. I can't make a release, if I could, I would have months ago - scikit-build-core has had to carry around over a year's worth of patches on pyproject-metadata now. I probably will sadly move to vending in the next scikit-build-core release.

File: CHANGELOG.rst [hunk 3]
```diff
@@ -2,6 +2,22 @@
...
 +++++++++
 
+0.8.0
+=====
+
+- Support specifying the `metadata_version` to 2.1, 2.2, and 2.3
+- Always normalize extras following PEP 685
+- Preserve the user-specified name style in the metadata. ``.canonical_name`` added to get the normalized name
+- Require "version" in the dynamic table if unset (following PEP 621)
+- Support extras using markers containing "or"
```
Comment (henryiii) [inline CHANGELOG.rst:12]:
```suggestion
- Support extras using markers containing "or"
- Support empty extras
```

Adding #95.

**Comment by pradyunsg** [comment]
@henryiii Could you resolve the merge conflicts with `main`?

**Comment by henryiii** [comment]
Sure, will rebase tonight.

**Comment by henryiii** [comment]
@pradyunsg rebased.

File: README.md [hunk 4]
```diff
@@ -1,10 +1,10 @@
...
-[![codecov](https://codecov.io/gh/FFY00/python-pyproject-metadata/branch/main/graph/badge.svg?token=9chBjS1lch)](https://codecov.io/gh/FFY00/python-pyproject-metadata)
-[![Documentation Status](https://readthedocs.org/projects/pyproject-metadata/badge/?version=latest)](https://pyproject-metadata.readthedocs.io/en/latest/?badge=latest)
+[![pre-commit.ci status](https://results.pre-commit.ci/badge/github/pypa/pyproject-metadata/main.svg)](https://results.pre-commit.ci/latest/github/pypa/pyproject-metadata/main)
+[![checks](https://github.com/pypa/pyproject-metadata/actions/workflows/checks.yml/badge.svg)](https://github.com/FFY00/python-pyproject-metadata/actions/workflows/checks.yml)
+[![tests](https://github.com/pypa/pyproject-metadata/actions/workflows/tests.yml/badge.svg)](https://github.com/pypa/pyproject-metadata/actions/workflows/tests.yml)
+[![codecov](https://codecov.io/gh/pypa/pyproject-metadata/branch/main/graph/badge.svg?token=9chBjS1lch)](https://codecov.io/gh/pypa/pyproject-metadata)
+[![Documentation Status](https://readthedocs.org/projects/pyproject-metadata/badge/?version=latest)](https://pyproject-metadata.pypa.io/en/latest/?badge=latest)
```
Comment (henryiii) [inline README.md:7]:
```suggestion
[![Documentation Status](https://readthedocs.org/projects/pyproject-metadata/badge/?version=latest)](https://pep621.readthedocs.io/en/latest/?badge=latest)
```

File: CHANGELOG.rst [hunk 5]
```diff
@@ -2,6 +2,23 @@
 Changelog
 +++++++++
 
+0.8.0
+=====
+
+- Support specifying the `metadata_version` to 2.1, 2.2, and 2.3
```
Comment (henryiii) [inline CHANGELOG.rst:8]:
```suggestion
- Support specifying the `metadata_version` as 2.1, 2.2, or 2.3
```

**Comment by henryiii** [comment]
I made an RC release. Everything seemed fine. Backend authors should check to make sure it works with backends. Everything happened automatically, which was nice. We could add approvals to the release environment, I think that would be nice, say two approvals from maintainers to publish?

**Comment by rgommers** [comment]
> We could add approvals to the release environment, I think that would be nice, say two approvals from maintainers to publish?

That does sound like a good idea to me.

**Pull request merged**